src/Symfony/Component/Config/Resource/FileResource.php

```diff
@@ -29,10 +29,16 @@ class FileResource implements SelfCheckingResourceInterface, \Serializable
      * Constructor.
      *
      * @param string $resource The file path to the resource
+     *
+     * @throws \InvalidArgumentException
      */
     public function __construct($resource)
     {
         $this->resource = realpath($resource);
+
+        if (false === $this->resource) {
+            throw new \InvalidArgumentException(sprintf('The file "%s" does not exist.', $resource));
```

Comment:
This change breaks our test suite. For example, the `ObjectRouteLoader` registers `FileResource` instances based on the filename returned by the `ReflectionClass`: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Routing/Loader/ObjectRouteLoader.php#L90 However, this fails when the object is a PHPUnit mock object, see https://travis-ci.org/symfony/symfony/jobs/106738857#L3760 (I am also not sure how this is working with classes stored in phar archives).